core/src/main/java/nu/yona/server/analysis/service/ActivityService.java

```diff
@@ -215,27 +221,26 @@ private long getTotalPageableItems(UserAnonymizedDto userAnonymized, ChronoUnit
 		UUID userAnonymizedId = userService.getUserAnonymizedId(userId);
 		UserAnonymizedDto userAnonymized = userAnonymizedService.getUserAnonymized(userAnonymizedId);
 
-		if (!userAnonymized.hasAnyBuddies())
-		{
-			return new PageImpl<>(Collections.emptyList(), pageable, 0);
-		}
-
 		Interval interval = getInterval(getCurrentDayDate(userAnonymized), pageable, ChronoUnit.DAYS);
-
 		Set<BuddyDto> buddies = buddyService.getBuddiesOfUserThatAcceptedSending(userId);
 		Set<UUID> userAnonymizedIds = buddies.stream().map(this::getBuddyUserAnonymizedId).collect(Collectors.toSet());
+		Set<UUID> activityCategoryIds = buddies.stream().map(BuddyDto::getGoals).flatMap(Set::stream)
```

Comment:
It would be good to explain here, e.g.:
`// Friend results are all included. Own results are only included if some friend has a goal in the same category.`